examples/log-source-app/linux/main.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/PlatformManager.h>
+
+#include <app/server/Server.h>
+#include <app/util/util.h>
+#include <credentials/DeviceAttestationCredsProvider.h>
+#include <credentials/examples/DeviceAttestationCredsExample.h>
+#include <lib/core/CHIPError.h>
+#include <lib/support/CHIPArgParser.hpp>
+#include <lib/support/CHIPMem.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+#include <log-source-common/DiagnosticLogsCommandHandler.h>
+
+#include <fstream>
+#include <iostream>
+#include <unistd.h>
+
+using chip::BitFlags;
+using chip::ArgParser::HelpOptions;
+using chip::ArgParser::OptionDef;
+using chip::ArgParser::OptionSet;
+using chip::ArgParser::PrintArgError;
+using chip::Messaging::ExchangeManager;
+
+// TODO: this should probably be done dynamically
+constexpr chip::EndpointId kOtaProviderEndpoint = 0;
+
+constexpr uint16_t kOptionFilepath = 'f';
+
+// Arbitrary BDX Transfer Params
+constexpr uint32_t kMaxBdxBlockSize                 = 1024;
+constexpr chip::System::Clock::Timeout kBdxTimeout  = chip::System::Clock::Seconds16(5 * 60); // Specification mandates >= 5 minutes
+constexpr chip::System::Clock::Timeout kBdxPollFreq = chip::System::Clock::Milliseconds32(500);
+
+// Global variables used for passing the CLI arguments to the DiagnosticLogsCommandHandler object
+const char * gOtaFilepath = nullptr;
+
```

Comment:
This is cut'n'paste of OTA examples that are already using some temporary ways of doing BDX session management. Strongly recommend against copying, and certainly recommend against keeping OTA constants  in here

Reply:
These are needed for log file transfer. But as long as it is not implemented yet, and these fields are not being used, I'll remove them now.